packages/coin-support-xrp/src/operations/createAccounts/index.ts

```diff
@@ -58,22 +59,36 @@ const getAddressesFromDevice: GetAddressesFromDevice<XrpApp> = async params => {
 
 const createAccountFromAddress: IMakeCreateAccountsObservableParams<XrpApp>['createAccountFromAddress'] =
   async (addressDetails, params) => {
-    const coin = xrpCoinList[params.coinId];
-    const name = `${coin.name} ${addressDetails.index + 1}`;
+    const { address, index, balance, txnCount, derivationPath, schemeName } =
+      addressDetails;
+    const { coinId, walletId } = params;
+
+    const coin = xrpCoinList[coinId];
+    const name = `${coin.name} ${index + 1}`;
+
+    let spendableBalance = new BigNumber(balance);
+    if (!spendableBalance.isZero()) {
+      const reserveBalance = await services.getAccountReserveBalance(
+        address,
+        coinId,
+      );
+      spendableBalance = spendableBalance.minus(reserveBalance);
+    }
```

Comment:
This can make spendableBalance negative, use `max(0,value)` instead of checking for zero.